course_catalog/etl/loaders.py

```diff
@@ -47,6 +139,11 @@ def load_program(program_data):
         object_id__in=[course.id for course in courses]
     ).delete()
 
+    if not created and not program.published:
+        search_task_helpers.delete_program(program)
+    else:
+        search_task_helpers.upsert_program(program)
```

Comment:
When an unpublished program is created, it will be saved to the search index.  Only when the import is run again will it be removed from the index.

Reply:
Similar logic for courses above

Reply:
Same is true for courses.